### DISCUSSION TITLE
Reset the progress bar?

### DISCUSSION THREAD

**Discussion by ebjacino** [jspsych/jsPsych#2381] [Q&A] [answered]
Hello, I'm making an experiment that has 2 sections in it, and I would like to have the progress bar increment until full for the first half, then reset to empty during the break, then increment until full for the second half. I have it incrementing until full, but can't get it to reset, and it just stays full for the whole second half. Any ideas? Thanks!

**Comment by jodeleeuw** [comment, marked as answer]
Hi @ebjacino !

You'll probably need to set the progress bar manually to do this. Have you seen the [docs for this](https://www.jspsych.org/7.1/overview/progress-bar/#manual-control)? I'm guessing that if you've got it to fill all the way for the first half you've figured this part out. Are you finding that the progress bar isn't resetting to 0 with something like `jsPsych.setProgressBar(0)`?

**Reply by ebjacino** [reply]
Hi! Yes I have this instruction block for the second half: 
```    var welcome = {
      type: "html-keyboard-response",
      stimulus: "Welcome to the next experiment. Press any key to begin.",
      on_start: function() {
        // set progress bar to 0 at the start of experiment
       jsPsych.setProgressBar(0);
        }
    };
```

And it remains full. I'm not getting any errors in the web developer tools.

**Reply by ebjacino** [reply]
Ok, after some more sleuthing I figured out the problem was with how I was pushing things to the timeline in the right order and having 2 different n_trials variables (one for first half and one for second half)! So I can confirm it is possible to reset the progress bar in the middle of the timeline.

**Reply by jodeleeuw** [reply]
Oh good! Glad you were able to figure out the source of the issue and also that the progress bar isn't buggy 😅
